pkg/gadgettracermanager/gadgettracermanager_test.go

```diff
@@ -299,4 +302,212 @@ func TestContainer(t *testing.T) {
 	if pid != 102 {
 		t.Fatalf("Error while looking up container2: unexpected pid %v", pid)
 	}
+
+	// Add new container with same pod and container name of container0 but in different namespace
+	respAddContainer, err := g.AddContainer(ctx, &pb.ContainerDefinition{
+		Id:        "abcde0-different",
+		Namespace: "another-namespace",
+		Podname:   "my-pod",
+		Name:      "container0",
+		Labels: []*pb.Label{
+			&pb.Label{Key: "key1", Value: "value1"},
+			&pb.Label{Key: "key2", Value: "value2"},
+		},
+	})
+	if err != nil {
+		t.Fatalf("Failed to add container: %v", err)
+	}
+	if respAddContainer == nil {
+		t.Fatal("Error while adding container: invalid response")
+	}
+
+	// Look up containers with label 'key1=value1'
+	selectedContainers := g.GetContainersBySelector(&pb.ContainerSelector{
+		Labels: []*pb.Label{
+			&pb.Label{Key: "key1", Value: "value1"},
+		},
+	})
+	if len(selectedContainers) != 1 {
+		t.Fatalf("Error while looking up containers by one label: invalid number of matches")
+	}
+	found := false
+	for _, l := range selectedContainers[0].Labels {
+		if l.Key == "key1" && l.Value == "value1" {
+			found = true
+			break
+		}
+	}
+	if !found {
+		t.Fatalf("Error while looking up containers by one label: unexpected container %+v",
+			selectedContainers[0])
+	}
+
+	// Look up containers with label 'key1=value1' and 'key2=value2'
+	selectedContainers = g.GetContainersBySelector(&pb.ContainerSelector{
+		Labels: []*pb.Label{
+			&pb.Label{Key: "key1", Value: "value1"},
+			&pb.Label{Key: "key2", Value: "value2"},
+		},
+	})
+	if len(selectedContainers) != 1 {
+		t.Fatalf("Error while looking up containers by multiple labels: invalid number of matches")
+	}
+	found = false
+	for _, l := range selectedContainers[0].Labels {
+		if (l.Key == "key1" && l.Value == "value1") || (l.Key == "key2" && l.Value == "value2") {
```

Comment:
I think it's not testing what you wanted to test. It's only checking if there is a key1=value1 **or** key2=value2 label. Not both of them.

Reply:
You are right, I fixed it. PTAL.

Reply:
WRT splitting up the function - I agree, I was suggesting the same to Alban on Friday. Once I will be done with the ongoing PRs, I will do it.